tests/export_back_compat_test.py

```diff
@@ -320,6 +324,27 @@ def test_cuda_eigh_cusolver_syev(self, dtype_name="f32", variant="syevj"):
     self.run_one_test(func, data, rtol=rtol, atol=atol,
                       check_results=partial(self.check_eigh_results, operand))
 
+  @parameterized.named_parameters(
+      dict(testcase_name=f"_dtype={dtype_name}_{variant}",
+           dtype_name=dtype_name, variant=variant)
+      for dtype_name in ("f32", "f64")
+      # We use different custom calls for sizes <= 32
+      for variant in ["syevj", "syevd"])
+  def test_rocm_eigh_hipsolver_syev(self, dtype_name="f32", variant="syevj"):
```

Comment:
Is this a copy of test_cuda_eigh_cusolver_syev? I am wondering if we can avoid have a copy. For example, would it work if you just use a conditional in the existing test to pick the right test data depending on the test platform?

Reply:
Yes, I was planning to do that as it makes sense, however for some reason I was getting `TypeError: CompatTest.test_cuda_eigh_cusolver_syev.<locals>.<lambda>() takes 0 positional arguments but 1 was given` whenever I use `eigh_harness` function. That's why I created my own function and decided to do a separate test!

Reply:
Can you please try harder to use the old function? I do not see a reason why it should not work. Note that the `CompatTest.eigh_harness` is written indeed to take zero inputs.

Reply:
Oops, sorry for the oversight. It turned out that we need to tweak the `starter_data` function before generating the data for test one. But, now, I created generic "gpu" functions for both rocm and cuda. Thanks!